blocks/library/embed/index.js

```diff
@@ -26,7 +27,33 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
+	for ( const patternIndex in patterns ) {
+		const pattern = new RegExp( '^https?:\/\/' + patterns[ patternIndex ] + '$', 'i' );
+		if ( url.match( pattern ) ) {
+			return true;
+		}
+	}
+	return false;
+};
+
+const findBlock = ( url ) => {
+	const findMatch = ( blocks ) => {
+		for ( const blockIndex in blocks ) {
```

Comment:
`for...in` should generally not be used for array iteration:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in

Further, the performance characteristics between the two largely favor a standard for loop:

https://jsperf.com/for-loop-vs-for-in-loop

For what you're doing here, I'd suggest a standard `for ( let i = 0; ...`.

Or a similar effect could be achieved with [`_.find`](https://lodash.com/docs/4.17.5#find) (there is a native [`Array#find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) but it is **not** polyfilled in Gutenberg):

```js
function findBlock( url ) {
	const blocks = [ ...common, ...others ];
	const block = find( blocks, ( block ) => {
		return matchesPatterns( url, block.settings.patterns );
	} );

	if ( ! block ) {
		return 'core/embed';
	}

	return block.name;
}
```